### ISSUE TITLE
Investigate how hard would be to use Apache libcloud to support many clouds

### ISSUE THREAD

**Issue by thrix** [psss/tmt#2035]
There is this library that supports an impressive amount of clouds:

https://packages.fedoraproject.org/pkgs/python-libcloud/python3-libcloud/

https://libcloud.readthedocs.io/en/stable/supported_providers.html#provider-matrix

It would be good to investigate if we could support it to include other clouds easily.

**Comment by happz** [comment]
The first Artemis prototype was build around libcloud, we got rid of it 3 years and 1 month ago :)

It was hard to make it play nicely with our code, hard to direct its logging to channels we wanted to use, and debugging was even worse in the async world of message brokers and tasks. But, things might be better now, it's been 3 years.

**Comment by thrix** [comment]
yeah, that is what I remember :) but seems it is still alive, so worth rechecking :)

**Comment by psss** [comment]
@thrix, still worth investigating?

**Comment by psss** [comment]
Dropping in favor of adding this into Artemis.
